### PR TITLE
[freshness-policies] Allow setting freshness policies when using graph-backed assets

### PR DESCRIPTION
**Pull request by OwenKephart**
### Summary & Motivation

https://github.com/dagster-io/dagster/issues/11677

### CODE REVIEW

File: python_modules/dagster/dagster/_core/definitions/assets.py [hunk 1]
```diff
@@ -298,6 +299,10 @@ def from_graph(
                 be associated with each of the output assets for this node. Keys are names of the
                 outputs, and values are dictionaries of metadata to be associated with the related
                 asset.
+            freshness_policy_by_ouptut_name (Optional[Mapping[str, FreshnessPolicy]]): Defines a
```
Comment (sryza) [inline python_modules/dagster/dagster/_core/definitions/assets.py:302]:
```suggestion
            freshness_policy_by_output_name (Optional[Mapping[str, FreshnessPolicy]]): Defines a
```

**Review by sryza — COMMENTED** [review]
Thoughts on specifying this by asset key? I feel like that might be the more natural way to think about it?

OTOH, I know that's a little bit inconsistent with metadata_by_output_name.

**Comment by OwenKephart** [comment]
@sryza I think of this setup as analogous to the AssetOut stuff, where everything is keyed by output name (including asset keys, metadata, and freshness policies)

**Review by sryza — APPROVED** [review]
Ok, I buy it

**Pull request merged**